epochStart/metachain/rewardsV2_test.go

```diff
@@ -1763,6 +1763,83 @@ func TestNewRewardsCreatorV2_CreateRewardsMiniBlocks2169Nodes(t *testing.T) {
 	require.Nil(t, err)
 }
 
+func TestRewardsCreatorV2_CreateRewardsMiniBlocksHeaderV3(t *testing.T) {
+	t.Parallel()
+
+	t.Run("should return ErrNilHeaderHandler because of nil header argument", func(t *testing.T) {
+		t.Parallel()
+
+		args := getRewardsCreatorV2Arguments()
+		rwd, err := NewRewardsCreatorV2(args)
+		require.Nil(t, err)
+
+		nbEligiblePerShard := uint32(400)
+		vInfo := createDefaultValidatorInfo(nbEligiblePerShard, args.ShardCoordinator, args.NodesConfigProvider, 100, defaultBlocksPerShard)
+		metaBlock := &block.MetaBlockV3{}
+		prevBlockExecResults := block.BaseMetaExecutionResult{
+			BaseExecutionResult: &block.BaseExecutionResult{},
+		}
+
+		_, err = rwd.CreateRewardsMiniBlocksHeaderV3(nil, vInfo, &metaBlock.EpochStart.Economics, &prevBlockExecResults)
```

Comment:
It's not really necessary to create all those arguments to check for nil input arguments.
You can just use
```
&block.MetaBlock{} / &block.MetaBlockV3{}
state.NewShardValidatorsInfoMap()
&block.Economics{}
&block.BaseMetaExecutionResult{}
```